repositories/MerchOrderRepository.ts

```diff
@@ -48,7 +49,18 @@ export class MerchOrderRepository extends BaseRepository<OrderModel> {
    * Gets all orders for a given user. Returns the order joined with its pickup event and user.
    */
   public async getAllOrdersForUser(user: UserModel): Promise<OrderModel[]> {
-    return this.repository.find({ user });
```

Comment:
Curious if this was here before previously instead of the full query so we don't go over the max payload size when sending all orders for a user?

Reply:
Oh yea that's probably why, also I overlooked that this query is only orders and excludes order items. I guess the question now becomes–do we always want linkedEvent info coming back with every single order? I assume yes since it seems like the frontend wants to show the linked event for order history. I can remove the `orderItem` joins from the query, but if there's a way to bake the linkedEvent join into the `find` that would be cleanest.

Reply:
Realized that the `find` does this magical join because the joins are specified in the OrderModel, as explicitly stated [here](https://github.com/acmucsd/membership-portal/blob/5c1bd1b076ffda633c35cfbb97cfe25df4411949/models/OrderModel.ts#L22), and with the `eager:true` option. I don't think we establish any relations between orders and linked events, so I adjusted this query at the repository layer instead